google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugProcessStateCollector.java

```diff
@@ -0,0 +1,96 @@
+/*
```

Comment:
Easier: I prefer to test smaller classes resulting in smaller test classes. To test just this method it's about 250 lines and CloudDebugGlobalPoller has way more functionality (which is currently not very well tested) which could lead to a huge test class.

Querying the states that are listening in the background does not depend on any fields/methods of CloudDebugGlobalPoller, hence I think querying them is independent of that class' state.
CloudDebugGlobalPoller indeed queries snapshots and snapshot locations, but how I think about CloudDebugProcessStateCollector is more like a repository functionality which currently has only one use case: returning the debug sessions that listen in the background. Maybe in the future we'll need others like: return all cloud debugger sessions, return all active cloud debugger sessions, etc. I'd put those into this class.